parsers/CL.py

```diff
@@ -193,26 +209,39 @@ def fetch_production(
 
     req = s.get(url, headers=headers)
     raw_data = req.json()["aggs"]
-    processed_data = production_processor_historical(raw_data)
+    historical_data = production_processor_historical(raw_data, enable_live_parser)
 
-    data = []
-    for production_data in processed_data:
-        dt = production_data.pop("datetime")
+    """The last 9 datapoints should be omitted because they usually are incomplete and shouldn't appear on the map."""
+    del historical_data[-9:]
 
-        datapoint = {
-            "zoneKey": zone_key,
-            "datetime": dt,
-            "production": production_data,
-            "storage": {
-                "hydro": None,
-            },
-            "source": "coordinador.cl",
-        }
+    production_list = ProductionBreakdownList(logger)
 
-        data.append(datapoint)
+    for event in historical_data:
+        dt = event.pop("datetime")
 
-    return data[:-9]
-    """The last 9 datapoints should be omitted because they usually are incomplete and shouldn't appear on the map."""
+        production_mix = ProductionMix()
+        for key, value in event.items():
+            production_mix.add_value(key, value)
+
+        production_list.append(
+            zoneKey=zone_key,
+            datetime=dt,
+            production=production_mix,
+            storage=None,
+            source="coordinador.cl",
+        )
+
+    return production_list.to_list()
+
+
+@refetch_frequency(timedelta(days=1))
+def fetch_production(
+    zone_key: ZoneKey = ZoneKey("CL-SEN"),
+    session: Session | None = None,
+    target_datetime: datetime | None = None,
+    logger: Logger = getLogger(__name__),
+) -> list[dict[str, Any]]:
+    return _fetch_production(zone_key, session, target_datetime, logger)
```

Comment:
What is the point of doing this and not just use the old function like before?

Reply:
The reason why I introduced the `_fetch_production` internal function is to be able to unit test the parser's code that processes live data. The newly added unit test `test_snapshot_live_data` calls the `_fetch_production` function with parameter `enable_live_parser` set to `True`.
So the code that was in the `fetch_production` function has been moved to the internal `_fetch_production` function, which also provides the `enable_live_parser` parameter.

I was myself not sure if to add or not the new test `test_snapshot_live_data` for the live parser, because as you mentioned it is currently disabled. However, as the parser code still supports it and since as part of this PR I had to also update that code, I decided that it would be better to have a unit test covering the live parser.